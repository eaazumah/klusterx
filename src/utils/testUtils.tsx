import { configureStore } from '@reduxjs/toolkit'
import '@testing-library/jest-dom'
import { render as rtlRender, renderHook as rtlRenderHook } from '@testing-library/react'
import 'jest-styled-components'
import React from 'react'
import { Provider } from 'react-redux'
import settings from '../redux/settingsSlice'
import { IReduxSate } from '../types/declarations'

interface Options {
  store?: any
  preloadedState?: Partial<IReduxSate>
}

interface OptionsWithRouter {
  store?: any
  preloadedState?: Partial<IReduxSate>
}

const render = (
  ui: React.ReactElement,
  {
    preloadedState,
    store = configureStore({
      preloadedState,
      reducer: { settings },
    }),
    ...options
  }: Options = {},
) => {
  const Wrapper = ({ children }: { children: React.ReactNode }) => {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...options })
}

const renderWithRouter = (
  ui: React.ReactElement,
  {
    preloadedState,
    store = configureStore({
      preloadedState,
      reducer: { settings },
    }),
    ...options
  }: OptionsWithRouter = {},
) => {
  const Wrapper = ({ children }: { children: React.ReactNode }) => {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...options })
}

const renderHook = <TProps, TResult>(
  callback: (props: TProps) => TResult,
  {
    mocks = [],
    preloadedState,
    store = configureStore({
      preloadedState,
      reducer: { settings },
    }),
    ...options
  }: any = {},
) => {
  const Wrapper = ({ children }: { children: React.ReactNode }) => {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRenderHook(callback, {
    ...options,
    wrapper: Wrapper,
  })
}

// re-export everything
export * from '@testing-library/react'
// override render method
export { render, renderHook, renderWithRouter }
