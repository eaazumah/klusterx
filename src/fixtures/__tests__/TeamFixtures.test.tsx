import { render, screen } from '../../utils/testUtils'

import TeamFixtures from '../TeamFixtures'

jest.mock('../../data/data.ts', () => ({
  __esModule: true,
  default: () => [
    { score: { 'Team A': 3, 'Team B': 1 }, date: '2021-05-05T14:00:00' },
    { score: { 'Team B': 2, 'Team C': 2 }, date: '2021-05-05T14:00:00' },
  ],
}))

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
  useParams: () => ({ teamName: 'Team A' }),
}))

describe('TeamFixtures', () => {
  test('should render team fixtures', () => {
    const { asFragment } = render(<TeamFixtures />)
    expect(asFragment).toMatchSnapshot()
  })

  test('should render team fixtures with data', () => {
    const { container } = render(<TeamFixtures />)
    expect(container).toMatchSnapshot()

    expect(screen.getByText('Team A Fixtures')).toBeInTheDocument()
  })
})
