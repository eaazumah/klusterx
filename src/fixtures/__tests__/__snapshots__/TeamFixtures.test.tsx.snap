// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamFixtures should render team fixtures 1`] = `[Function]`;

exports[`TeamFixtures should render team fixtures with data 1`] = `
.c1 .fixture {
  margin: 15px;
  margin-left: 30px;
  margin-right: 30px;
}

.c1 .badge {
  text-transform: none;
}

.c0 {
  padding-top: 100px;
}

<div>
  <div
    class="mantine-Container-root c0 mantine-1pdibkf"
  >
    <div
      class="mantine-Group-root mantine-k3ov3c"
    >
      <button
        class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-ayz740"
        type="button"
      >
        <svg
          class="tabler-icon tabler-icon-arrow-left"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12l14 0"
          />
          <path
            d="M5 12l6 6"
          />
          <path
            d="M5 12l6 -6"
          />
        </svg>
      </button>
      <div
        class="mantine-Badge-root mantine-1q104z4"
      >
        <span
          class="mantine-1t45alw mantine-Badge-inner"
        >
          Team A
           Fixtures
        </span>
      </div>
    </div>
    <div
      class="mantine-1cw9ra3"
    />
    <div
      class=""
    >
      <h1
        class="mantine-Text-root mantine-Title-root mantine-99mnmj"
      >
        Wednesday 05 May 2021
      </h1>
      <div
        class="mantine-Divider-root mantine-Divider-horizontal mantine-urkk9t"
        role="separator"
      />
      <div
        class="c1"
      >
        <div
          class="fixture mantine-1jggmkl"
        >
          <div
            class="mantine-Group-root mantine-7ftt1"
          >
            <a
              class="mantine-Text-root mantine-bwhnbm"
              href="/fixtures/Team A"
            >
              Team A
            </a>
            <div
              class="mantine-Text-root mantine-4fbajm"
            >
              vs
            </div>
            <a
              class="mantine-Text-root mantine-bwhnbm"
              href="/fixtures/Team B"
            >
              Team B
            </a>
          </div>
          <div
            class="mantine-1gn90f5"
          />
          <div
            class="mantine-Group-root mantine-7ftt1"
          >
            <div
              class="mantine-Text-root mantine-1d564l0"
            >
              3
               - 
              1
            </div>
          </div>
          <div
            class="mantine-1gn90f5"
          />
          <div
            class="mantine-Text-root mantine-1d564l0"
          >
            14:00
          </div>
        </div>
        <div
          class="mantine-Divider-root mantine-Divider-horizontal mantine-urkk9t"
          role="separator"
        />
      </div>
      <div
        class="mantine-1cw9ra3"
      />
    </div>
  </div>
</div>
`;
